src/app/clusters/door-lock-server/door-lock-server.cpp

```diff
@@ -380,11 +380,10 @@ void DoorLockServer::setUserCommandHandler(chip::app::CommandHandler * commandOb
         return;
     }
 
-    if (!userType.IsNull() &&
-        (userType.Value() < UserTypeEnum::kUnrestrictedUser || userType.Value() > UserTypeEnum::kRemoteOnlyUser))
+    if (!userType.IsNull() && userType.Value() == UserTypeEnum::kUnknownEnumValue)
```

Comment:
would `userType == Optional::Value(UserTypeEnum::kUnknownEnumValue)` work instead of double-compare?

Reply:
There is no static named `Value` on Optional (or Nullable, which is the actual type of `UserType`).

That said, we could do `userType == MakeNullable(UserTypeEnum::kUnknownEnumValue)`.

Reply:
Or even better, add an `operator==` on Nullable that takes `T` refs.